waku/v2/node/wakunode2.nim

```diff
@@ -1013,9 +1013,34 @@ when isMainModule:
     
     ok(storeTuple)
 
-  # 2/6 Initialize node
+  # 2/7 Retrieve dynamic bootstrap nodes
+  proc retrieveDynamicBootstrapNodes(conf: WakuNodeConf): Result[seq[RemotePeerInfo], cstring] =
```

Comment:
Does this block? One concern would be if the node gets busy discovering nodes and isn't able to start functioning earlier

Reply:
Yes it blocks. Agreed.
The blocking stems form [waku_dnsdisc](https://github.com/status-im/nim-waku/blob/e6fdbdb42ac8592c02811d4a2709d4d17e065d0c/waku/v2/node/dnsdisc/waku_dnsdisc.nim#L52).
Imo, we should address this in a separate issue.

For now, this should not be too severe, because it is "just" blocking DNS queries. These DNS records should be available in the next setup step and would be waited for anyways.
I think the DNS disc API should be async, but the setup functions should be sync because the steps sequentially depend on each other.

Should I open a new issue or take further steps within this PR?

Reply:
Agree and opening a new issue sounds good, thanks!